src/DotVVM.Framework/ViewModel/Serialization/IViewModelServerStore.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Threading.Tasks;
+using DotVVM.Framework.Runtime.Caching;
+
+namespace DotVVM.Framework.ViewModel.Serialization
+{
+    public interface IViewModelServerStore
+    {
+
+        void Store(string hash, string cacheData);
```

Comment:
Good idea.

Reply:
Newtonsoft.Json has support for BSON, but in 12.0 and older versions it's included in the Newtonsoft.Json package itself. From 12.0 it is in the separate package. Since DotVVM tries to request the lowest Newtonsoft.Json as possible, it will be difficult to make this binary compatible. Maybe we can use Reflection to find the BsonWriter and BsonReader class.

Reply:
And btw ReadOnlyMemory is only in .NET Standard 2.1 which means we cannot use this for .NET Framework. I will use byte array and BSON - it should be more efficient than parsing and writing the string.